### ISSUE TITLE
Member <anything> may be missing on module `torch.distributed`

### ISSUE THREAD

**Issue by arogozhnikov** [astral-sh/ty#2244]
### Summary

```python
import torch.distributed as tdist

tdist.all_gather # complains on all ops I use: get_rank / get_world_size / ...
```

while this seemingly works

```
from torch.distributed import ... 
```

### Version

ty 0.0.7

**Comment by MichaReiser** [comment]
I'm not sure what the issue is exactly but all other type checkers that I tested (Pylance, Pyrefly) don't raise the same diagnostic.

**Comment by carljm** [comment]
We have already disabled `possibly-unresolved-reference` and `possibly-missing-import` by default, due to the likelihood of false positives when libraries do complex conditional imports. (The latter being disabled by default is I think the only reason `from torch.distributed import ...` shows no issue here.) So for consistency I think we should also disable `possibly-missing-attribute` by default, otherwise you get this weird inconsistency where the import version is fine but the attribute version is not.

**Comment by AlexWaygood** [comment]
> for consistency I think we should also disable `possibly-missing-attribute` by default, otherwise you get this weird inconsistency where the import version is fine but the attribute version is not.

We cannot do that without first tackling https://github.com/astral-sh/ty/issues/1656

**Comment by kmontemayor2-sc** [comment]
Hi it seems like #1656 is resolved now, though I see the same issues when I'm trying to setup `ty 0.0.17` in [our codebase](https://github.com/Snapchat/GiGL). 

[e.g.](https://github.com/Snapchat/GiGL/blob/main/tests/unit/nn/models_test.py#L123): 

```
warning[possibly-missing-attribute]: Member `destroy_process_group` may be missing on module `torch.distributed`
   --> tests/unit/nn/models_test.py:123:25
    |
121 |             rank=0, world_size=1, init_method=get_process_group_init_method()
122 |         )
123 |         self.addCleanup(torch.distributed.destroy_process_group)
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Note that we're currently using `mypy 1.8.0` and it passes these checks in our codebase.

**Comment by AlexWaygood** [comment]
We probably also want to change this error to use a different error code to `possibly-missing-attribute` before disabling `possibly-missing-attribute` by default: https://github.com/astral-sh/ruff/blob/410902fa401afda969cc000f13be341896e6868e/crates/ty_python_semantic/src/types/infer/builder.rs#L13502-L13526

**Comment by carljm** [comment]
@kmontemayor2-sc 
The planned fix here is to simply disable `possibly-missing-attribute` rule by default (since it highlights a potential error that other type checkers don't even try to highlight, and is prone to false positives), so the easy workaround for now is to do that yourself in your `ty.toml`:

```toml
[rules]
possibly-missing-attribute = "ignore"
```

**Comment by kmontemayor2-sc** [comment]
Thanks @carljm :)
